.github/workflows/build.yml

```diff
@@ -308,6 +308,19 @@ jobs:
           docker build --target go-export -f Dockerfile.proto -o type=local,dest=node .
           git diff --name-only --exit-code && echo "✅ Generated proto matches committed proto" || (echo "❌ Generated proto differs from committed proto, run \`rm -rf node/pkg/proto && docker build --target go-export -f Dockerfile.proto -o type=local,dest=node .\` and commit the result" >&2 && exit 1)
 
+  spellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Pinned version of the v6 tag, which is a lightweight and hence mutable tag
+      - uses: streetsidesoftware/cspell-action@214db1e3138f326d33b7a6a51c92852e89ab0618
+        with:
+          # For now, only lint markdown files
+          files: '**/*.md'
+          inline: warning
+          # Only check files in the PR or push
+          incremental_files_only: true
```

Comment:
It is not necessary, but I was cognizant of the performance aspect given existing slow build times. It is just doing extra work.

If we ever opt to enable this for **/*.js and **/*.ts (which I did test!), it will take significantly longer. I just figured there is no need to do extra work when it isn't necessary.